### DISCUSSION TITLE
Paid Web Requests

### DISCUSSION THREAD

**Discussion by WebShells** [opensourcepos/opensourcepos#3668] [Team Posts]
Dears,
Hope everyone is doing well... 
Been discussing with @Jekkos regarding several ways to get additional income either on the project side or coding side... To share a brief overview, we agreed to offer custom development for features or special requests for people who are willing to pay for it based on hourly rate to be agreed on.
If anybody have the time or planning to get few additional money, we are really getting a good number of requests. (You'll see once i share the accounts, i just remembered that i didn't add yet...)
Another thing is cloud based development, still figuring out what's the best to be done here... 
Please let me know if you agree upon this:
Average/Hour of custom feature development is 100 Euro.
Requests to be processed by an interval of 24 to 72 hours. 
We have the right to accept the task or not.
Please i need your opinions about this... 
Thanks all for the continuous work everyday you are putting here. 
Shady

**Comment by SteveIreland** [comment]
Tempting.

**Comment by daN4cat** [comment]
The tool has potential, it's quite appealing now and offers a good set of features. Go for that, I'm perfectly fine for you to make money on the extra work.

Question: once a paid implementation is done, will you share it to the public branch or are you considering keeping it as a custom commercial branch in a private GitHub?
If the latter, I wouldn't mind to be able to access the code if possible.

With regards to cloud based development, what do you mean with that? A SaaS & multitenant solution?
If you need any help with deployments on Kubernetes, let me know as I worked on that in the past year and using Azure AKS.

**Comment by jekkos** [comment]
Personally I'm not that much interested in doing custom dev myself, but a friend of mine would (he's running a PHP web development company). The idea was to forward some of these feature requests to his company and try to agree on some sort of approach in which they would contribute that work to the project upstream.

But in short I don't really think that a lot of people are willing to pay something decent for the work that is delivered. Let's see how it goes.

On the other hand I was thinking on setting up a kubernetes cluster. In short I want to upgrade the VPS infrastructure I have now for demo etc to a managed k8s. The automation done there could then be used commercially through a simple signup page for external people.

1. Manually migrate demo and personal instances to a cluster (probably with MySQL on a VPS at first)
2. Automate deployment by pulling builds from docker hub and installing them as pods in k8s
3. Subdomain registration mechanism that then points to the newly deployed pod
4. Mysql could be deployed by modifying the configmap to point to AWS Aurora or DO managed Mysql. This means databases have automated backups and recovery out of the box.
5. In the end this service can be exposed through a simple webpage for external people that can sign up for a monthly or yearly fee for the hosting and updates/backups.
6. Different types of deployments (commercial/dev) can be separated using k8s namespaces

@daN4cat as you say that would be a multitenant solution as every user will have his own subdomain, pod and database schema. DigitalOcean now has a managed k8s offering which looks easy to start with. I think having something like that for dev purposes alone would be very useful.

**Comment by WebShells** [comment]
Your feedbacks are awesome ❤️... 
@daN4cat we can divide this into 2 options, either we start a parallel repo to apply all custom changes to it and offer it as a paid version of our release (This will push curious users to buy monthly based plan in order to gain what's additional in the project. Or we can simply get enough out of it and release it to public over our main repository. Both options are doable, but for sure to be contributed back! (Here am thinking business wise, that will be an opportunity to offer something very rich for additional 10 or 15 usd/month per setup added to the base price of the package).
Kubernetes is a great option as @jekkos mentioned and digital oceans are offering a webinar as an introduction, ill leave this for you guys..  What am trying to do here is what we got by now is rich and appealing as dan said, stable as per my personal experience... Problems are very limited to minor fixes, reliable, & trustful app overall. Giving it the right social or online exposure will push it forward definitely, let's move to business then 😎. 
Cheers 🍺

**Comment by objecttothis** [comment]
Sorry, I'm quite late to the discussion.  I only just figured out that the org could have private discussion here.  I'm contributing to the project because it benefits the bookstore that my wife and I own.  This means that most of my contributions are directly beneficial to our efforts but I still try to write them so that they benefit the project as a whole... this is mostly because I don't want to maintain a separate repository that I have to keep up to date with this one... but also I like being a team player ;)

With that said, if there are project requests that I am already planning to implement in some fashion, I'd be glad to consider it.  I do think it would be nice if any projects that get taken on by the org have the right to be merged into the master for others to use, but perhaps that's asking too much.

**Comment by WebShells** [comment]
@objecttothis we have been discussing that all the additional requests will be merged up in a paid version to be released over the web too... 
So we have 2 different solutions here, the free community ( github for users who are willing to search, read, ask, contribute, etc... ) and web for people who wants to get their work done without doing any effort, like: requests, paid features based on hourly rates, managed cloud based POS, etc...
Either we offer all the additions that were requested through the web for free and add them back to Github, or we offer all as a quarterly PAID cloud based managed POS solution. 
Please let's decide about it, i'll create all the accounts now..

**Comment by WebShells** [comment]
@opensourcepos/org Please all confirm that you received an email from OSPOS with your accounts/password.
Thank you.

**Comment by odiea** [comment]
Yes I got it. Will try to figure out what I can use as a 12 character password.

**Comment by odiea** [comment]
It was already created. Thanks

**Comment by objecttothis** [comment]
> @objecttothis we have been discussing that all the additional requests will be merged up in a paid version to be released over the web too...
> So we have 2 different solutions here, the free community ( github for users who are willing to search, read, ask, contribute, etc... ) and web for people who wants to get their work done without doing any effort, like: requests, paid features based on hourly rates, managed cloud based POS, etc...
> Either we offer all the additions that were requested through the web for free and add them back to Github, or we offer all as a quarterly PAID cloud based managed POS solution.
> Please let's decide about it, i'll create all the accounts now..

Thank you for sharing.  Offering a cloud-hosted version of the software should generate some revenue.  Having a community edition vs full-featured paid edition could also help, although maintaining two repository sets could be a pain.  If you don't maintain two then you need to do enough obfuscation to prevent potential users from just enabling the paid features with coding changes.  You will have users who want to pay for the features but want to self-host.

I recommend with starting by offering just one codeset, paid cloud-hosting and paid feature requests that get merged into the codeset.  Then down the road if you wanted to divide a community edition and paid edition you could do that.  Making all the changes at once is perhaps problematic.

This is potentially unrelated, but it would also help to have a published roadmap for the software.  Right now, it's kind of at the mercy of whims of contributors (not always a bad thing) and then whatever @jekkos and @daN4cat  have planned, but not published.

**Comment by objecttothis** [comment]
Also, this might already be in the works, but I do want to request that core contributors have access to the full version (if that's the road that you go down) without paying for it.

**Comment by WebShells** [comment]
@objecttothis all the team members/collaborators will have access to all features, web, and git repo as needed... None of us will be limited and the revenue will be shared among all of us... That's to be fair :)... Was discussing this with @jekkos...

**Comment by objecttothis** [comment]
> @objecttothis all the team members/collaborators will have access to all features, web, and git repo as needed... None of us will be limited and the revenue will be shared among all of us... That's to be fair :)... Was discussing this with @jekkos...

Makes sense.  Is the plan to first take the overhead costs of hosting, accounts, etc out of the revenue and then disperse the remaining among the members then?  Also, I assume that you are talking about the cloud-hosting and full version revenue.  I think revenue generated from developers implementing requested features should go to the developer that implemented the feature.  Perhaps you were already thinking along those lines.

**Comment by WebShells** [comment]
> I think revenue generated from developers implementing requested features should go to the developer that implemented the feature.

@objecttothis for sure, the assigned tasks per developer are the developer's rights... He have the rights to fully take that amount... But the revenue generated from the cloud managed solutions that will result from the whole project will be divided as mentioned. 
Every one of us is putting a great effort in his field to this project, and am not done here... I guess we are able to make this go more viral, more trending, more used worldwide... That will be something i'll be happy to get in challenge with and take care of... Cheers and Good luck!
